src/nested_pandas/nestedframe/core.py

```diff
@@ -289,38 +405,48 @@ def from_lists(cls, df, base_columns=None, list_columns=None, name="nested"):
-        expr = _ensure_spacing(expr)
-        nest_exprs = {col: [] for col in self.nested_columns + ["base"]}  # type: dict
-        split_expr = expr.split(" ")
-
-        i = 0
-        current_focus = "base"
-        while i < len(split_expr):
-            expr_slice = split_expr[i].strip("()")
-            # Check if it's a nested column
-            if self._is_known_hierarchical_column(expr_slice):
-                nested, colname = split_expr[i].split(".")
-                current_focus = nested.strip("()")
-                # account for parentheses
-                j = 0
-                while j < len(nested):
-                    if nested[j] == "(":
-                        nest_exprs[current_focus].append("(")
-                    j += 1
-                nest_exprs[current_focus].append(colname)
-            # or if it's a top-level column
-            elif expr_slice in self.columns:
-                current_focus = "base"
-                nest_exprs[current_focus].append(split_expr[i])
-            else:
-                nest_exprs[current_focus].append(split_expr[i])
-            i += 1
-        return {expr: " ".join(nest_exprs[expr]) for expr in nest_exprs if len(nest_exprs[expr]) > 0}
+    def eval(self, expr: str, *, inplace: bool = False, **kwargs) -> Any | None:
+        """
+
+        Evaluate a string describing operations on NestedFrame columns.
+
+        Operates on columns only, not specific rows or elements.  This allows
+        `eval` to run arbitrary code, which can make you vulnerable to code
+        injection if you pass user input to this function.
+
+        Works the same way as `pd.DataFrame.eval`, except that this method
+        will also automatically unpack nested columns into NestedSeries,
+        and the resulting expression will have the dimensions of the unpacked
+        series.
+
+        Parameters
+        ----------
+        expr : str
+            The expression string to evaluate.
+        inplace : bool, default False
+            If the expression contains an assignment, whether to perform the
+            operation inplace and mutate the existing NestedFrame. Otherwise,
+            a new NestedFrame is returned.
+        **kwargs
+            See the documentation for :func:`eval` for complete details
+            on the keyword arguments accepted by
+            :meth:`~pandas.NestedFrame.eval`.
 
-    def query(self, expr) -> Self:  # type: ignore[name-defined] # noqa: F821
+        Returns
+        -------
+        ndarray, scalar, pandas object, nested-pandas object, or None
+            The result of the evaluation or None if ``inplace=True``.
+
+        See Also
+        --------
+        https://pandas.pydata.org/docs/reference/api/pandas.eval.html
+        """
+        kwargs["resolvers"] = tuple(kwargs.get("resolvers", ())) + (_NestResolver(self),)
+        kwargs["inplace"] = inplace
+        kwargs["parser"] = "nested-pandas"
+        return super().eval(expr, **kwargs)
```

Comment:
It can return `_SeriesFromNest` on occasion, true.  This is a descendant of `pd.Series` already, so expressions like `instanceof(obj, pd.Series)` will be true for it.  I can't think of an easy way to change it to `pd.Series` (and nothing more) without impacting the ability for `query` to do the right thing; it uses this type tag to decide what level of querying to perform.